sdk/core/azure-core/src/test/java/com/azure/core/implementation/serializer/HttpResponseBodyDecoderTests.java

```diff
@@ -130,8 +128,10 @@ public <T> T deserialize(InputStream inputStream, Type type, SerializerEncoding
             new UnexpectedExceptionInformation(HttpResponseException.class));
 
         HttpResponse response = new MockHttpResponse(GET_REQUEST, 300);
+        Object deserializedResponse =
+            HttpResponseBodyDecoder.decodeByteArray(null, response, ioExceptionThrower, noExpectedStatusCodes);
 
-        assertNull(HttpResponseBodyDecoder.decodeByteArray(null, response, ioExceptionThrower, noExpectedStatusCodes));
+        assertTrue(deserializedResponse instanceof IOException);
```

Comment:
This should use `assertInstanceOf` as that will produce a better exception message if this fails.